src/main/java/com/mastery/java/task/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,190 @@
+            if (out != 0) {
+                System.out.println("Update Employee with id=" + employee.getEmployeeId());
+            } else System.out.println("Employee update failed with id=" + employee.getEmployeeId());
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public void deleteById(int id) {
+        String query = "DELETE from employee where employee_id=?";
+        Connection con = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            con = dataSource.getConnection();
+            preparedStatement = con.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            int out = preparedStatement.executeUpdate();
+            if (out != 0) {
+                System.out.println("Employee deleted with id=" + id);
+            } else System.out.println("No Employee found with id=" + id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public List<Employee> getAll() {
+        String query = "SELECT employee_id, first_name, last_name, departament_id, job_title, gender, date_of_birth from employee";
+        List<Employee> employeeList = new ArrayList<Employee>();
+        Connection con = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        try {
+            con = dataSource.getConnection();
+            preparedStatement = con.prepareStatement(query);
+            resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Employee employee = new Employee();
+                employee.setEmployeeId(resultSet.getInt("employee_id"));
+                employee.setFirstName(resultSet.getString("first_name"));
+                employee.setLastName(resultSet.getString("last_name"));
+                employee.setDepartamentId(resultSet.getInt("departament_id"));
+                employee.setJobTitle(resultSet.getString("job_title"));
+                employee.setGender(Gender.valueOf(resultSet.getString("gender")));
+                employee.setDateOfBirth(resultSet.getDate("date_of_birth"));
+                System.out.println("Employee Found::" + employee);
+                employeeList.add(employee);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        } finally {
+            try {
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
```

Comment:
Why do we suppress this exception instead to handle it?

Reply:
Done